test/test_dependency_versions.py

```diff
@@ -0,0 +1,142 @@
+"""
+A unit test to ensure that the versions of dependency packages stay in sync
+between open-source builds of Sketcher and the builds used in the Schrodinger
+core suite.
+
+Note that this test should **not** be launched from premake or the generated
+Makefile. Buildinger strips $SCHRODINGER_PACKAGE_FACTORY_BASE from the
+environment when it executes those, which means that the unit test may not be
+able to find the pixi executable, which it needs to fetch version information.
+The test should instead be launched directly from a terminal via::
+
+    $SCHRODINGER/utilities/py.test test_dependency_versions.py
+"""
+
+import itertools
+import json
+import logging
+import os
+import pathlib
+import re
+import sys
+import types
+from collections.abc import Iterable
+
+SKETCHER_TEST_DIR = pathlib.Path(__file__).parent
+SKETCHER_VERSION_FILE = SKETCHER_TEST_DIR / ".." / "external" / "versions.json"
+# The name of the package factory virtual environment to use when fetching
+# version info from mmshare
+MMSHARE_PF_ENVIRONMENT = "schrodinger"
+# Packages that we should exclude from the version checks
+PACKAGES_TO_IGNORE = [
+    # Core suite builds frequently use a commercial-only LTS release of Qt,
+    # which are not available for open-source builds of Sketcher
+    "qt",
+    # The core suite doesn't include emscripten
+    "emscripten",
+    # this repo is currently on a newer RDKit and Boost versions than the core
+    # suite
+    "rdkit",
```

Comment:
oh! I updated rdkit in https://schrodinger.atlassian.net/browse/SKETCH-2426 -- this is one of the ones I want a report that it's out of sync so I remember to update when Ricardo updates in the suite